File: src/main/java/it/polito/ai/pedibus/api/models/Line.java
package it.polito.ai.pedibus.api.models;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import it.polito.ai.pedibus.api.constraints.ValidTimeArray;
import it.polito.ai.pedibus.api.utils.ValidList;
import it.polito.ai.pedibus.api.serializers.ObjectIdSerializer;
import lombok.Data;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.validation.Valid;
import javax.validation.constraints.Email;

@Data
@ValidTimeArray
@Document(collection = "lines")
public class Line {
    @Id
    @JsonSerialize(using = ObjectIdSerializer.class)
    private ObjectId id;
    private String name;
    @Valid
    private ValidList<Stop> stops;
    @Email
    private String admin_email;
}

File: frontend/src/app/models/line.ts
import {Stop} from "./stop";
import {StopList} from "./stop-list";
import {Builder} from "builder-pattern";
import {LocalTime} from "js-joda";

export class Line {
  name;
  id;
  admin_email;
  stops: StopList;

  constructor(values: any = {}) {
    this.name = values.name;
    this.id = values.id;
    this.admin_email = values.admin_email;
    this.stops = new StopList(values.stops);
  }
}
